src/tokens/tokens.interfaces.ts

```diff
@@ -0,0 +1,201 @@
+// Copyright © 2022 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ApiProperty, OmitType } from '@nestjs/swagger';
+import { IsNotEmpty, IsOptional } from 'class-validator';
+
+// Ethconnect interfaces
+export interface EthConnectAsyncResponse {
+  sent: boolean;
+  id: string;
+}
+
+export interface EthConnectReturn {
+  output: string;
+}
+
+// REST API requests and responses
+export class AsyncResponse {
+  @ApiProperty()
+  id: string;
+}
+
+export enum TokenType {
+  FUNGIBLE = 'fungible',
+}
+
+const contractConfigDescription =
+  'Optional configuration info for the token pool. Reserved for future use.';
+const requestIdDescription =
+  'Optional ID to identify this request. Must be unique for every request. ' +
+  'If none is provided, one will be assigned and returned in the 202 response.';
+
+export class TokenPool {
+  @ApiProperty()
+  @IsNotEmpty()
+  name: string;
+
```

Comment:
You do need to accept a "type" param here, because FireFly is going to pass one. It should be optional, and you default to "fungible" if not specified. If anything other than "fungible" or empty string is given, consider it an error.

Reply:
Since all ERC20 contracts are fungible, my thinking was that we just ignore `type` for this connector. In FF, the user must specify what connector they want to use when creating a pool, and I think that should be the deciding factor on how we handle `type`. For example, if user specifies a connector that uses both fungible/nonfungible like ERC1155, type should not be ignored. For connectors that strictly have one type, it should be ignored. Thoughts?

Reply:
In FF, the same `fftokens` plugin will be used to talk to erc1155 or erc20. FireFly has no way of knowing what's on the other side (and should not need to care) unless we introduce additional config parameters. That's why we're trying to keep the API consistent between the erc1155 and erc20 connectors, so we don't have to tailor a new FireFly plugin for each new connector.

Reply:
got it, will add the error checks